app/src/main/java/com/google/android/soundchecker/BaseFilePlayerActivity.kt

```diff
@@ -160,31 +164,28 @@ open class BaseFilePlayerActivity : ComponentActivity(), OnAudioFocusChangeListe
                 .setWillPauseWhenDucked(false)
                 .setOnAudioFocusChangeListener(this, mHandler)
                 .build()
-    }
-
-    override fun onResume() {
-        super.onResume()
-        becomingNoisyReceiver = BecomingNoisyReceiver()
-        registerReceiver(becomingNoisyReceiver,
-                IntentFilter(AudioManager.ACTION_AUDIO_BECOMING_NOISY))
-    }
-
-    override fun onStart() {
-        super.onStart()
+        checkAndRequestNotificationPermission()
         val intent = Intent(this, FilePlayerService::class.java)
-        intent.putExtra(FilePlayerService.TITLE, mTag)
-        intent.putExtra(FilePlayerService.SONG_NAME, mFileName)
         startForegroundService(intent)
-    }
 
-    override fun onPause() {
-        unregisterReceiver(becomingNoisyReceiver)
-        super.onPause()
+        registerReceiver(mBecomingNoisyReceiver,
+                IntentFilter(AudioManager.ACTION_AUDIO_BECOMING_NOISY))
+
+        val playbackStateIntentFilter = IntentFilter(FilePlayerService.ACTION_START_PLAYBACK)
+        playbackStateIntentFilter.addAction(FilePlayerService.ACTION_STOP_PLAYBACK)
+        registerReceiver(mPlaybackStateChangedReceiver, playbackStateIntentFilter,
+                         RECEIVER_EXPORTED)
     }
 
     override fun onDestroy() {
+        unregisterReceiver(mBecomingNoisyReceiver)
+        unregisterReceiver(mPlaybackStateChangedReceiver)
         mAudioManager.abandonAudioFocusRequest(mFocusRequest)
+        stopPlayback()
         mMsgHandlerThread.quitSafely()
+        val intent = Intent(this, FilePlayerService::class.java)
+        intent.action = FilePlayerService.ACTION_STOP_SERVICE
+        startForegroundService(intent)
```

Comment:
This looks weird to me but I believe it is correct.

Reply:
Yes, that is correct. By calling `startForegoundService`, `onStartCommand` will be called. The intent is handled there. If the service is not running, `onCreate` for the service will be first called and then `onStartCommand` will be called after that.